File: Data Structures & Algorithms/max-water-container/submission-1.java
class Solution {
    public int maxArea(int[] heights) 
    {
       int r = heights.length-1;
       int l = 0;
       int res = 0;

       while(l<r) 
       {
            res = Math.max(res,(r-l)*Math.min(heights[r],heights[l]));
            if(heights[l]<heights[r])
            {
                l++;
            }
            else
            {
                r--;
            }

       }
       return res;
    }
}

File: Data Structures & Algorithms/reverse-integer/submission-1.py
class Solution:

    def reverse(self, x):

        MIN = -2147483648
        MAX = 2147483647

        result = 0

        sign = -1 if x < 0 else 1

        x = abs(x)

        while x:

            digit = x % 10

            x //= 10

            result = result * 10 + digit

        result *= sign

        if result < MIN or result > MAX:
            return 0

        return result